cmd/fdo_client/client.go

```diff
@@ -342,11 +346,30 @@ func di() (err error) { //nolint:gocyclo
 	default:
 		return fmt.Errorf("unsupported key encoding: %s", diKeyEnc)
 	}
+
+	var deviceInfo string
+	switch {
+	case diDeviceInfo != "" && diDeviceInfoMac != "":
+		return fmt.Errorf("can't specify both -di-device-info and -di-device-info-mac")
+	case diDeviceInfo != "":
+		deviceInfo = diDeviceInfo
+	case diDeviceInfoMac != "":
+		deviceInfo, err = getMac(diDeviceInfoMac)
+		if err != nil {
+			return fmt.Errorf("error getting device information from iface %s: %w", diDeviceInfoMac, err)
+		}
+	default:
+		deviceInfo, err = getSerial()
```

Comment:
This will:

1. Always error on non-Linux systems
2. Always error on Linux systems without elevated privileges

Therefore we are effectively making the `di-device-info/di-device-info-mac` flags required outside of the privileged Linux case.

I'm okay with this. It enforces that deviceinfo is not just some default string.

Reply:
yup, for now that's the case, maybe on windows/mac there will be other means to gather a meaningful value there
for the privileges need on linux, the error makes it clear you need root so I'm less worried (tpm access requires root as well unless you chmod it properly, at least from my testing on my own laptop)